app/code/Magento/CompareListGraphQl/Model/Service/CompareListService.php

```diff
@@ -0,0 +1,272 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CompareListGraphQl\Model\Service;
+
+use Magento\Catalog\Block\Product\Compare\ListCompare;
+use Magento\Catalog\Helper\Product\Compare;
+use Magento\Catalog\Model\CompareList;
+use Magento\Catalog\Model\Config as CatalogConfig;
+use Magento\Catalog\Model\Product;
+use Magento\Catalog\Model\Product\Visibility as CatalogProductVisibility;
+use Magento\Catalog\Model\ResourceModel\CompareList as ResourceCompareList;
+use Magento\Catalog\Model\ResourceModel\Product\Compare\Item\Collection;
+use Magento\CatalogGraphQl\Model\Resolver\Product\Price\Discount;
+use Magento\CatalogGraphQl\Model\Resolver\Product\Price\ProviderPool as PriceProviderPool;
+use Magento\Framework\GraphQl\Query\Resolver\ContextInterface;
+use Magento\Catalog\Model\ResourceModel\Product\Compare\Item\CollectionFactory as CompareItemsCollectionFactory;
+use Magento\Framework\Pricing\SaleableInterface;
+use Magento\Store\Api\Data\StoreInterface;
+
+
+class CompareListService
+{
+    /**
+     * @var Collection
+     */
+    private $items;
+
+    /**
+     * @var CompareItemsCollectionFactory
+     */
+    private $itemCollectionFactory;
+
+    /**
+     * @var CatalogProductVisibility
+     */
+    private $catalogProductVisibility;
+
+    /**
+     * @var CatalogConfig
+     */
+    private $catalogConfig;
+
+    /**
+     * @var Compare
+     */
+    private $compareProduct;
+
+    /**
+     * @var Discount
+     */
+    private $discount;
+
+    /**
+     * @var PriceProviderPool
+     */
+    private $priceProviderPool;
+
+    /**
+     * @var ResourceCompareList
+     */
+    private $resourceCompareList;
+
+    /**
+     * @var CompareList
+     */
+    private $modelCompareList;
+
+    /**
+     * @var ListCompare
+     */
+    private $blockListCompare;
+
+    /**
+     * @param CompareItemsCollectionFactory $itemCollectionFactory
+     * @param CatalogProductVisibility      $catalogProductVisibility
+     * @param CatalogConfig                 $catalogConfig
+     * @param Compare                       $compareHelper
+     * @param PriceProviderPool             $priceProviderPool
+     * @param Discount                      $discount
+     * @param ResourceCompareList           $resourceCompareList
+     * @param CompareList                   $compareList
+     * @param ListCompare                   $listCompare
+     */
+    public function __construct(
```

Comment:
This service contains more dependencies than recommended for a single class as well. Also, I see that there are two entry points (public functions). In order to follow the Separation of Concerns principle I would strongly recommend splitting this service into (at least) two services with one entry point per service correspondingly.